### PR TITLE
Availability controller

### LINKED ISSUE #32: BE: Availability Controllers GET /availability
Create a route that: 
- Loads availability for given day (GET /availability?day=2342343243) 
- Returns a list of available slots 

Google Calendar API
- /freeBusy timeMin, timeMax 

Each date you select, it’ll load the appointment slots available

[
  {start_time: 2323234, end_time: 234234},
  {start_time: 2323234, end_time: 234234},
  {start_time: 2323234, end_time: 234234},
  ...
]

### PR DESCRIPTION
**Pull request by jk-me**
- passed basic test cases for algorithm
- endpoint to freebusy to algorithm response flow created

### CODE REVIEW

File: server/utils/availabilityHelper.js [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    } else if (endFirst && curr.isSameOrAfter(userStart)) {
+      if (busy && busyStart.isBefore(end)) {
+        addToSlots(curr, busyStart, reqMeet, slotTime, slots);
+        curr = busyEnd;
+        b++;
+      } else {
+        addToSlots(curr, end, reqMeet, slotTime, slots);
+        curr = end;
+      }
+    } else if (!endFirst && curr.isBetween(userStart, userEnd, null, '[)')) {
```
Comment (bonnieli) [inline server/utils/availabilityHelper.js:228]: let's add some comments for each of these if statements explaining what it's comparing - especially for this comparison for `[}` :)

File: server/utils/availabilityHelper.js [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+        curr = end;
+      }
+    } else {
+      if (curr.isBefore(userStart)) {
+        curr = userStart;
+      } else break;
+    }
+  }
+  let clientSlots = slots.map((time) => moment.tz(time, clientTz).format());
+  return clientSlots;
```
Comment (bonnieli) [inline server/utils/availabilityHelper.js]:
```suggestion
  return slots.map((time) => moment.tz(time, clientTz).format());
```

File: server/utils/availabilityHelper.js [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+        curr = busyEnd;
+        b++;
+      } else {
+        addToSlots(curr, userEnd, reqMeet, slotTime, slots);
+        curr = end;
+      }
+    } else {
+      if (curr.isBefore(userStart)) {
+        curr = userStart;
+      } else break;
```
Comment (bonnieli) [inline server/utils/availabilityHelper.js]: don't need the `else break;` here

**Review by bonnieli — COMMENTED** [review]
very cool - would love to see some more comments around each of the helper functions, and the smaller calculations in each func. in 1 month, or if you pass on your work to your teammates, they could pick it up pretty quickly!

File: server/utils/availabilityHelper.js (hunk 3 above)
Comment (jk-me) [reply server/utils/availabilityHelper.js]: > very cool - would love to see some more comments around each of the helper functions, and the smaller calculations in each func. in 1 month, or if you pass on your work to your teammates, they could pick it up pretty quickly!

Thanks Bonnie! I've added more comments, and the changes you suggested :)

**Pull request merged**